### PR TITLE
375 allow imix to compile to a windows service 2

### LINKED ISSUE #375: Allow imix to compile to a windows service.
**Is your feature request related to a problem? Please describe.**
Windows requires a special service exe to run as a service.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
- Allows imix to compile as a windows service exe
- Refactors imix main entry into a service and standard 
- Moves main_standard and main_loop to lib
- Refactor reflective loader include bytes

#### Which issue(s) this PR fixes:
Fixes #375

### CODE REVIEW

File: implants/imix/src/main.rs [hunk 1]
```diff
@@ -1,229 +1,42 @@
...
-                Ok(_) => {}
-                Err(local_err) => {
-                    eprintln!(
-                        "An error occured during installation: {}",
-                        local_err.to_string()
-                    )
-                }
-            };
+    #[cfg(all(feature = "win_service", windows))]
+    match service_dispatcher::start("myservice", ffi_service_main) {
```
Comment (Cictrone) [inline implants/imix/src/main.rs:22]: maybe a different name 😭

File: implants/imix/src/win_service.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+use std::{ffi::OsString, time::Duration};
+
+#[cfg(all(feature = "win_service", windows))]
+fn run_service() -> windows_service::Result<()> {
+    use windows_service::{
```
Comment (Cictrone) [inline implants/imix/src/win_service.rs:5]: 🙁 why is the import in the function?

File: implants/imix/src/win_service.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        match control_event {
+            ServiceControl::Stop | ServiceControl::Interrogate => {
+                ServiceControlHandlerResult::NoError
+            }
+            _ => ServiceControlHandlerResult::NotImplemented,
+        }
+    };
+
+    // Register system service event handler
+    let status_handle = service_control_handler::register("my_service_name", event_handler)?;
```
Comment (Cictrone) [inline implants/imix/src/win_service.rs:23]: the name 😭

File: implants/imix/src/win_service.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        service::{
+            ServiceControl, ServiceControlAccept, ServiceExitCode, ServiceState, ServiceStatus,
+            ServiceType,
+        },
+        service_control_handler::{self, ServiceControlHandlerResult},
+    };
+
+    let event_handler = move |control_event| -> ServiceControlHandlerResult {
+        match control_event {
+            ServiceControl::Stop | ServiceControl::Interrogate => {
```
Comment (Cictrone) [inline implants/imix/src/win_service.rs:15]: will restart work?

**Review by Cictrone — DISMISSED** [review]
just a few 😭

File: docs/_docs/user-guide/imix.md [hunk 5]
```diff
@@ -85,7 +85,7 @@ def main():
 main()
 ```
 
-Installation scripts are specified in the `realm/implants/imix/install_scripts` directeroy.
+Installation scripts are specified in the `realm/implants/imix/install_scripts` directory.
```
Comment (jabbate19) [inline docs/_docs/user-guide/imix.md:88]: Spelling is hard sometimes

**Review by jabbate19 — APPROVED** [review]
LGTM

**Comment by hulto** [comment]
Gotta update to the new refactor

**Pull request closed without merging**